source/BannerlordCoopInstaller/Installer.cs

```diff
@@ -0,0 +1,264 @@
+﻿using Microsoft.Win32;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Drawing;
+using System.IO;
+using System.Reflection;
+using System.Resources;
+using System.Text;
+using System.Windows.Forms;
+using System.Runtime.InteropServices;
+using System.Runtime.InteropServices.ComTypes;
+
+namespace BannerlordCoopInstaller
+{
+    public partial class BannerlordCoopInstallerForm : Form
+    {
+
+        private Dictionary<string, ResourceFile> resourceDict;
+
+        /// <summary>
+        /// Check the BannerlordPath and ensure and executable exists. If not, disable the install button
+        /// </summary>
+        private void checkBannerLordPath()
+        {
+            if(File.Exists(this.BannerlordPath.Text + "\\bin\\Win64_Shipping_Client\\Bannerlord.exe"))
+            {
+                this.Install.Enabled = true;
+                this.Status.Text = "Ready to install";
+            }
+            else
+            {
+                this.Status.Text = "Select a Mount && Blade II: Bannerlord install path to continue";
+                this.Install.Enabled = false;
+            }
+        }
+
+        /// <summary>
+        /// Check if MB is installed (through the registry) - Only works in x64.
+        /// </summary>
+        /// <returns>Full path where MB is installed</returns>
+        private string GetBannerLordInstalledPath()
+        {
+
+            string registry_key = @"SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall";
+            using (Microsoft.Win32.RegistryKey key = Registry.LocalMachine.OpenSubKey(registry_key))
+            {
+                foreach (string subkey_name in key.GetSubKeyNames())
```

Comment:
This feels a bit unnecessary. Can we use LINQ to find the subkey we're looking for? Do we only expect a few subkeys in each key?  Probably fine since it's just during install...